SimPEG/utils/model_utils.py

```diff
@@ -1,10 +1,30 @@
-from .mat_utils import mkvc
+import warnings
+from typing import Literal, Optional
+
+import discretize
 import numpy as np
-from scipy.interpolate import griddata
-from scipy.spatial import cKDTree
 import scipy.sparse as sp
 from discretize.utils import active_from_xyz
-import warnings
+from scipy.interpolate import griddata
+from scipy.spatial import cKDTree
+
+from .mat_utils import mkvc
+
+try:
+    from numba import njit, prange
+except ImportError:
+    # Define dummy njit decorator
+    def njit(*args, **kwargs):
+        return lambda f: f
+
+    # Define dummy prange function
+    prange = range
+
+    warnings.warn(
+        "numba is not installed. Some computations might be slower.",
+        type=ImportWarning,
+        stacklevel=2,
+    )
```

Comment:
I would move this warning to the function body, otherwise every user will receive it after importing SimPEG. Moreover, I would be more explicit on what computations might be slower and maybe point people to instructions on how to install Numba.
 
To move it to the function's body you could include an `import numba` in the `try` statement and during the exception assign it to `None`:

```python
try:
    import numba
    from numba import njit, prange
except ImportError:
    numba = None
    # Define dummy njit decorator
    def njit(*args, **kwargs):
        return lambda f: f

    # Define dummy prange function
    prange = range
```

And then in the function:

```python

def depth_weighting(
    mesh, reference_locs, active_cells=None, exponent=2.0, threshold=None, **kwargs
):
    if numba is None:
        warnings.warn(...)
```

Reply:
Done!
Note: i removed the `ImportWarning` because it is ignored by default (thus it would not have shown except in debug mode)

Reply:
Oh, yes. We shouldn't raise an `ImportWarning` in this case, just a `UserWarning` should be fine.